test/core/2_pendleData.spec.ts

```diff
@@ -30,20 +41,191 @@ describe('PendleData', async () => {
+    await expect(data.setExpiryDivisor(BN.from(divisor))).to.be.revertedWith("INVALID_EXPIRY_DIVISOR");
+  })
+
+  it('Onlyforge modifier should revert transactions from non-forge', async()=> {
+    await expect(data.connect(alice).storeTokens(consts.FORGE_AAVE_V2, fixture.a2Forge.a2OwnershipToken.address, fixture.a2Forge.a2FutureYieldToken.address, tokens.USDT.address, consts.T0_A2.add(consts.SIX_MONTH))).to.be.revertedWith("ONLY_FORGE");
+  })
+
+  it('Should be able to add forge', async() => {
+    const av2ForgeAddress = await data.getForgeAddress(consts.FORGE_AAVE_V2);
+    expect(av2ForgeAddress).to.be.eq(fixture.a2Forge.aaveV2Forge.address);
+    const cForgeAddress = await data.getForgeAddress(consts.FORGE_COMPOUND);
+    expect(cForgeAddress).to.be.eq(fixture.cForge.compoundForge.address);
+  })
+
+  it('Should revert invalid forge addition', async() => {
+    await expect(data.addForge(consts.ZERO_BYTES, consts.RANDOM_ADDRESS)).to.be.revertedWith("ZERO_BYTES");
+    await expect(data.addForge(consts.FORGE_COMPOUND, consts.ZERO_ADDRESS)).to.be.revertedWith("ZERO_ADDRESS");
+    await expect(data.addForge(consts.FORGE_COMPOUND, fixture.a2Forge.aaveV2Forge.address)).to.be.revertedWith("INVALID_ID");
+    await expect(data.addForge(consts.FORGE_AAVE_V2, fixture.a2Forge.aaveV2Forge.address)).to.be.revertedWith("EXISTED_ID");
+  })
+
+  it('Should be able to set forge fee', async() => {
+    await expect(data.setForgeFee(consts.RONE.div(10))).to.emit(data, "ForgeFeeSet").withArgs(consts.RONE.div(10));
+    const forgeFeeInData = await data.forgeFee();
+    expect(forgeFeeInData).to.be.eq(consts.RONE.div(10));
+  })
+
+  it('Should revert setting forge fee above cap', async() =>{
+    await expect(data.setForgeFee(consts.RONE)).to.be.revertedWith("FEE_EXCEED_LIMIT");
+  });
+
+  it('Should be able to return correct pendle yield tokens', async() => {
+    const yieldTokens = await data.getPendleYieldTokens(consts.FORGE_AAVE_V2, tokens.USDT.address, consts.T0_A2.add(consts.SIX_MONTH));
+    const [ot, xyt] = yieldTokens;
+    expect(ot).to.be.eq(fixture.a2Forge.a2OwnershipToken.address);
+    expect(xyt).to.be.eq(fixture.a2Forge.a2FutureYieldToken.address);
+  });
+
+  it('Should be able to correctly determine validity of xyt', async() => {
+    const resExpectTrue = await data.isValidXYT(consts.FORGE_AAVE_V2, tokens.UNI.address, consts.T0_A2.add(consts.SIX_MONTH));
+    expect(resExpectTrue).to.be.true;
+    const resExpectFalse = await data.isValidXYT(consts.FORGE_COMPOUND, tokens.USDT.address, consts.T0_C.add(consts.ONE_YEAR));
+    expect(resExpectFalse).to.be.false;
+  })
+
+  it('Should be able to correctly determine validity of ot', async() => {
+    const resExpectTrue = await data.isValidOT(consts.FORGE_COMPOUND, tokens.USDT.address, consts.T0_C.add(consts.SIX_MONTH));
+    expect(resExpectTrue).to.be.true;
+    const resExpectFalse = await data.isValidOT(consts.FORGE_AAVE_V2, tokens.USDC.address, consts.T0_C.add(consts.SIX_MONTH));
```

Comment:
These variables should be named "isValidOT". Or you can just put the (await promise) into the expect function.